### PR TITLE
feat(extension): lw-8600

### PR DESCRIPTION
**Pull request by vetalcore**
# Checklist

- [x] JIRA - [LW-8600](https://input-output.atlassian.net/browse/LW-8600)
- [x] Proper tests implemented
- [x] Screenshots added.

# TODO:
- [x] align with [figma](https://www.figma.com/file/RPN3V4Et7hMbXkV5ZIF75i/Lace-Browser---Full-view-UI-Design?type=design&node-id=27311%3A127424&mode=design&t=lvsxRrdLIWp5ZsKt-1), which has next tx types: `Vote Signing`, `DRep Registration`, `Drep De-Registration`, `Vote Delegation`, `Revoke Delegation`, `Update Vote Delegation`.
- [x] tx data is mocked, so wasn't able to test details drawer due to some failures while fetching additional assets data etc.

---

## Proposed solution

Apply proper mapping on the tx inspection lvl, add handlers for the activities sections ui.

## Testing

Describe here, how the new implementation can be tested.
Provide link or briefly describe User Acceptance Criteria/Tests that need to be met

## Screenshots

Screenshot 2024-01-06 at 21 00 22
Screenshot 2024-01-10 at 12 17 15
Screenshot 2024-01-10 at 12 57 48
Screenshot 2024-01-10 at 12 58 29
Screenshot 2024-01-10 at 13 15 48

[LW-8600]: https://input-output.atlassian.net/browse/LW-8600?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/core/src/ui/components/ActivityDetail/ActivityTypeIcon.tsx [hunk 1]
```diff
@@ -1,43 +1,69 @@
...
 
-  return icon ? <Icon style={iconStyle} component={icon} /> : <QuestionOutlined style={iconStyle} />;
+  // TODO: set fill color for every icon using currentColor, replace icon not to contain surrounding circle
+  return icon ? (
+    <Icon
+      style={iconStyle}
+      // Override fill color for governance related transactions
+      className={cn({
+        [styles.governance]:
+          Object.values(ConwayEraCertificatesTypes).includes(type as unknown as ConwayEraCertificatesTypes) ||
```
Comment (mchappell) [inline packages/core/src/ui/components/ActivityDetail/ActivityTypeIcon.tsx:61]: for readability please extract this to a boolean value before the jsx return

**Review by mchappell — CHANGES_REQUESTED** [review]

File: apps/browser-extension-wallet/src/utils/tx-inspection.ts [hunk 2]
```diff
@@ -68,26 +132,38 @@ export const inspectTxType = ({
...
-        return 'delegation';
+        return DelegationTransactionType.delegation;
       case inspectionProperties.stakeKeyRegistration.length > 0:
-        return 'delegationRegistration';
+        return DelegationTransactionType.delegationRegistration;
       case inspectionProperties.stakeKeyDeregistration.length > 0:
-        return 'delegationDeregistration';
+        return DelegationTransactionType.delegationDeregistration;
+      // Voting procedures take priority over proposals
+      // TODO: should we create proper inspector instead?
```
[the remark this answers is not in the document]
Comment (vetalcore) [reply apps/browser-extension-wallet/src/utils/tx-inspection.ts:154]: done https://input-output.atlassian.net/browse/LW-9569 
could you please help me to create proper ticket for the implementation itself?

File: apps/browser-extension-wallet/src/utils/tx-inspection.ts [hunk 3]
```diff
@@ -112,3 +188,41 @@ export const inspectTxValues = ({
...
+}
+
+export const getVoterType = (voterType: Wallet.Cardano.VoterType): VoterTypeEnum => {
+  switch (voterType) {
+    case Wallet.Cardano.VoterType.ccHotKeyHash:
+    case Wallet.Cardano.VoterType.ccHotScriptHash:
+      return VoterTypeEnum.CONSTITUTIONAL_COMMITTEE;
+    case Wallet.Cardano.VoterType.stakePoolKeyHash:
+      return VoterTypeEnum.SPO;
+    case Wallet.Cardano.VoterType.dRepKeyHash:
```
[the remark this answers is not in the document]
Comment (vetalcore) [reply apps/browser-extension-wallet/src/utils/tx-inspection.ts:205]: done

File: packages/core/src/ui/components/ActivityDetail/ActivityTypeIcon.tsx [hunk 4]
```diff
@@ -1,43 +1,69 @@
...
+  [ConwayEraCertificatesTypes.ResignCommitteeCold]: DelegationIcon,
+  [ConwayEraCertificatesTypes.AuthorizeCommitteeHot]: DelegationIcon
 };
 
 export const ActivityTypeIcon = ({ type }: ActivityTypeIconProps): React.ReactElement => {
   const icon = type && activityTypeIcon[type];
   const iconStyle = { fontSize: txIconSize() };
 
-  return icon ? <Icon style={iconStyle} component={icon} /> : <QuestionOutlined style={iconStyle} />;
+  // TODO: set fill color for every icon using currentColor, replace icon not to contain surrounding circle
```
[the remark this answers is not in the document]
Comment (vetalcore) [reply packages/core/src/ui/components/ActivityDetail/ActivityTypeIcon.tsx:54]: ticket has been created

File: packages/core/src/ui/components/ActivityDetail/ActivityTypeIcon.tsx (hunk 1 above)
Comment (vetalcore) [reply packages/core/src/ui/components/ActivityDetail/ActivityTypeIcon.tsx:61]: done

File: apps/browser-extension-wallet/src/utils/mocks/governance.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
+import { Wallet } from '@lace/cardano';
+
+// const STAKE_KEY_HASH = Wallet.Cardano.RewardAccount.toHash(REWARD_ACCOUNT);
```
[the remark this answers is not in the document]
Comment (vetalcore) [reply apps/browser-extension-wallet/src/utils/mocks/governance.ts:3]: done

File: apps/browser-extension-wallet/src/stores/slices/activity-detail-slice.ts [hunk 6]
```diff
@@ -205,10 +219,13 @@ const buildGetActivityDetail =
       addrOutputs: outputs,
       metadata: txMetadata,
       includedUtcDate: blocks?.utcDate,
-      includedUtcTime: blocks?.utcTime
+      includedUtcTime: blocks?.utcTime,
+      votingProcedures: votingProceduresTransformer(tx.body.votingProcedures),
```
[the remark this answers is not in the document]
Comment (vetalcore) [reply apps/browser-extension-wallet/src/stores/slices/activity-detail-slice.ts:223]: done

**Review by mchappell — APPROVED** [review]

**Pull request merged**